baseimage/Makefile

```diff
@@ -0,0 +1,233 @@
+	ln -sf /root/.composer/vendor/bin/drush /usr/bin/drush
+	@echo "Drush installed."
+
+# Installs the drush-launcher instead of drush.
+.PHONY: install-drush-launcher
+install-drush-launcher: install-package-wget
+	$(info Installing Drush Launcher...)
+	wget -O /tmp/drush.phar https://github.com/drush-ops/drush-launcher/releases/download/0.6.0/drush.phar
+	mv -f /tmp/drush.phar /usr/bin/drush
+	chmod +x /usr/bin/drush
+	@echo "Drush Launcher installed."
+
+# Install terminus for Pantheon sites.
+.PHONY: install-terminus
+install-terminus: install-composer install-package-rsync
+	$(info Installing Terminus...)
+	curl -O https://raw.githubusercontent.com/pantheon-systems/terminus-installer/master/builds/installer.phar
+	php installer.phar install
+	@echo "Terminus installed."
+
+# Install a specific version of PHP by passing the version to this target. Use
+# major and minor versions separated by a dot. For example, install-php-7.2.
+.PHONY: install-php-%
+install-php-%:
+	$(info Installing PHP $*...)
+#	# Ensure PHP version is correctly formatted.
+	@if [[ ! $* =~ ^[0-9]\.[0-9]$$ ]]; then\
+		echo "PHP version $* is an invalid format.";\
+		exit 1;\
+	fi
+#	# Ensure installed PHP version isn't the current version to be installed.
+	@if [ $* = ${current_php_version} ]; then\
+		echo "Current PHP Version is already ${current_php_version}.";\
+	else \
+		apt-get install -y python-software-properties software-properties-common;\
+		LC_ALL=C.UTF-8 add-apt-repository -y ppa:ondrej/php;\
+		apt-get update;\
+		apt-get install -y \
+			php$(*) \
+			php$(*)-mbstring \
+			php$(*)-mysql \
+			php$(*)-xml \
+			php$(*)-zip \
+			php$(*)-bcmath \
+			php$(*)-bz2 \
+			php$(*)-cli \
+			php$(*)-common \
+			php$(*)-curl \
+			php$(*)-dev \
+			php$(*)-gd \
+			php$(*)-intl \
+			php$(*)-json \
+			php$(*)-mbstring \
+			php$(*)-mysql \
+			php$(*)-opcache \
+			php$(*)-phpdbg \
+			php$(*)-pspell \
+			php$(*)-readline \
+			php$(*)-recode \
+			php$(*)-soap \
+			php$(*)-sqlite3 \
+			php$(*)-tidy \
+			php$(*)-xml \
+			php$(*)-xsl \
+			php$(*)-zip \
+			libapache2-mod-php$(*);\
+		a2enmod php$(*);\
+		a2dismod php$(current_php_version) || /bin/true;\
+		echo "PHP $(*) installed.";\
+	fi
+
+# Install a specific version of nodejs by passing the version to this target.
+# Use the major version only. For example, install-nodejs-9.
+.PHONY: install-nodejs-%
+install-nodejs-%: install-package-curl
+	$(info Installing nodejs $*...)
+#	# Ensure nodejs version is correctly formatted.
+	@if [[ ! $* =~ ^[0-9]$$ ]]; then\
+		echo "Version number $* is an invalid format. Specify the major version only.";\
+		exit 1;\
+	fi
+	curl -sL https://deb.nodesource.com/setup_$(*).x | bash -
+	$(MAKE) install-package-nodejs
+	@echo "Installed nodejs $(*).x."
+
+# Clean up apt repos etc., to reduce disk size.
+.PHONY: clean
+clean:
+	apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Test this Makefile
+.PHONY: test
+test:
```

Comment:
Hmm, I probably need to rename this since someone may already have a `test` target in their makefile.

Reply:
`clean` is also a commonly used target